returnn/tf/layers/base.py

```diff
@@ -559,6 +559,10 @@ def _set_global_batch_by_data(data):
         # Just leave the placeholder as-is. Most layers should anyway reset this.
         output.placeholder = x
     if out_shape is not None:
+      if output.batch:
+        for dim in output.batch.virtual_dims:
+          if isinstance(dim, BatchInfo.PackedDim):
+            out_shape.remove(dim.dim_tag)
```

Comment:
How else would you remove the corresponding dim tag in the `out_shape`? I could not think of a way within `_make_layer` because we dont always have `BatchInfo` there that gives us info which axis is packed.

Reply:
Anyway, this change is wrong. You cannot change this here. This is wrong in general. This can only be done in `_make_layer` because only then you can know that it is correct.

Reply:
From my understanding there should never be a `PackedDim` within `out_shape` when `output` has `BatchInfo`, thats why I thought this should be safe here, but I see that the preferred way would be to handle it within `_make_layer`.

Reply:
> From my understanding there should never be a `PackedDim` within `out_shape` when `output` has `BatchInfo`

Why not? (Btw, `PackedDim` is just a wrapper by `BatchInfo`. It wraps a dim tag. `out_shape` is just about dim tags.)

Reply:
Because a `PackedDim` is packed in the batch, so the dim should not be in the dim_tags (and also is not, it is just not removed from the out_shape of the layer)

Reply:
> Why not? Nothing is stopping you from having it that way.

Because this is how I understand the packing to work right now. But if you say the packed dim tag is not always removed from `dim_tags` when packing I agree that this does not work. I will change it to removing `out_shape`.

Reply:
Ah you are right. I did not think of this case.